### PR TITLE
Added examples into reports

### PR DESCRIPTION
**Pull request by gano3s**
I one more time slightly enhanced the report. 

1. I added a capability to load images from attachments to each vulnerability and also to the project.
1.1 Vulnerabilities and images
image
image
1.2 Project attachments (only images for now)
image
image

2. I updated how PoC of vulnerability is propagated into the ouput. I am using MD parser to generate the HTML, but I am not inserting it directly into the PHPWord as in case of description. Instead of it, I use XPath to extract all `<code>` elements and I parse these manually. 
This approach has one benefit, PoC can have more examples and all of them are in the output and parsed differently. The con is, that it strips another elements. 

image
image

# generate doc and template
[template.docx](https://github.com/reconmap/rest-api/files/8418386/template.docx)
[test.docx](https://github.com/reconmap/rest-api/files/8418387/test.docx)

### CODE REVIEW

File: src/Controllers/Reports/CreateReportController.php [hunk 1]
```diff
@@ -29,6 +29,23 @@ public function __construct(
...
+    		$resultarray=array();
+    		foreach ($elements as $element) {
+    		    $nodes = $element->childNodes;
+    		    foreach ($nodes as $node) {
+    		      $resultarray[] = $node->nodeValue;
+    		    }
+    		}
+    		return $resultarray;
+    	}
+    }
```
Comment (santiagolizardo) [inline src/Controllers/Reports/CreateReportController.php:46]: This looks like a nice tiny method to extract into its own class. I'll do it after merge 👍🏼

File: src/Services/Reporting/ReportDataCollector.php [hunk 2]
```diff
@@ -60,6 +60,17 @@ public function collectForProject(int $projectId): array
             $vulnerability_sort = "FIELD(v.owasp_overall, 'critical','high','medium','low','note')";
         }
         $vulnerabilities = $this->vulnerabilityRepository->search($vulnerabilitySearchCriteria, null, $vulnerability_sort);
+        foreach ($vulnerabilities as $key => $vuln) {
+            $pngs = $projectAttachments = $this->attachmentRepository->findByParentId('vulnerability', $vuln['id'], 'image/png');
+            $jpegs = $projectAttachments = $this->attachmentRepository->findByParentId('vulnerability', $vuln['id'], 'image/jpeg');
+            $gifs = $projectAttachments = $this->attachmentRepository->findByParentId('vulnerability', $vuln['id'], 'image/gif');
```
Comment (santiagolizardo) [inline src/Services/Reporting/ReportDataCollector.php:66]: I think we should refactor this later to be 
```php
$images = $this->attachmentRepository->findByParentId('vulnerability', $vuln['id'], 'image/*'); // All images
```

File: src/Services/Reporting/ReportDataCollector.php [hunk 3]
```diff
@@ -60,6 +60,17 @@ public function collectForProject(int $projectId): array
             $vulnerability_sort = "FIELD(v.owasp_overall, 'critical','high','medium','low','note')";
         }
         $vulnerabilities = $this->vulnerabilityRepository->search($vulnerabilitySearchCriteria, null, $vulnerability_sort);
+        foreach ($vulnerabilities as $key => $vuln) {
+            $pngs = $projectAttachments = $this->attachmentRepository->findByParentId('vulnerability', $vuln['id'], 'image/png');
```
Comment (santiagolizardo) [inline src/Services/Reporting/ReportDataCollector.php:64]: I think you forgot to remove the `$projectAttachments` variable :)

**Review by santiagolizardo — APPROVED** [review]

**Comment by santiagolizardo** [comment]
Great job @gano3s !

File: src/Services/Reporting/ReportDataCollector.php (hunk 2 above)
Comment (gano3s) [reply src/Services/Reporting/ReportDataCollector.php:66]: Oh, great idea. It is more elegant than my solution. Thank you!

File: src/Services/Reporting/ReportDataCollector.php (hunk 3 above)
Comment (gano3s) [reply src/Services/Reporting/ReportDataCollector.php:64]: you are right. Thank you!

**Pull request merged**